flink-core/src/main/java/org/apache/flink/core/memory/HybridMemorySegment.java

```diff
@@ -306,6 +307,9 @@ public final void get(int offset, ByteBuffer target, int numBytes) {
 		if ((offset | numBytes | (offset + numBytes)) < 0) {
 			throw new IndexOutOfBoundsException();
 		}
+		if (target.isReadOnly()) {
```

Comment:
Isn't this check redundant? Shouldn't the `ByteBuffer`s validate it on their own like `DirectByteBufferR` do? Putting this check here, it complicates the code and we have to pay for it on each call, even on happy path (cost should be very tiny). 

Maybe it should be put only in `if (target.isDirect())` branch to check `read-only` only before unsafe copy?

Reply:
you are right - the non-direct buffers path is based on `ByteBuffer#array()` which will throw a `ReadOnlyBufferException` for read-only buffers, so it really is enough in the direct buffers code path where the `UNSAFE.copyMemory` is not checking the source pointer (how should it?!)